.github/workflows/build-stable.yml

```diff
@@ -94,33 +55,81 @@ jobs:
-      uses: ncipollo/release-action@v1
+    outputs:
+      version: ${{ steps.release_info.outputs.version }}
+
+  build:
+    name: Build artifacts
+
+    needs: release_info
+
+    strategy:
+      matrix:
+        include:
+          - { platform: macos-latest, rid: osx-arm64 }
+          - { platform: macos-latest, rid: osx-x64 }
+          - { platform: ubuntu-latest, rid: linux-arm }
+          - { platform: ubuntu-latest, rid: linux-arm64 }
+          - { platform: ubuntu-latest, rid: linux-x64 }
+          - { platform: windows-latest, rid: win-x64 }
+
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v3
       with:
-        artifacts: Celeste64-Fuji-${{ runner.os }}-${{ matrix.arch }}.zip
-        tag: "stable-${{ env.SHORT_SHA }}"
-        body: |
-          # Instructions:
+        fetch-tags: true
 
-          - Download and extract the zip file below according to your computer
-          - Run the Celeste64-Fuji application from the extracted files
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 8.0.x
 
-          # Requirements:
+    - name: Build
+      run: >
+        dotnet publish Celeste64.Launcher/Celeste64.Launcher.csproj
+          -c Release
+          -r ${{ matrix.rid }}
+          -p:ImportByWildcardBeforeSolution=false
+          -o build
+          && cp -r Content Mods build
 
-          - **Windows:** 10 or later, x64
-          - **Linux:** [Distro support list](https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md), x64 pr arm
-          - **macOS:** Monterey or later, x64 or arm Intel-based or Apple Silicon with Rosetta
-        allowUpdates: true
-        name: "Fuji ${{ env.SHORT_SHA }}"
-        prerelease: false
-        makeLatest: true
-        draft: true
-        commit: "${{ github.sha }}"
+    - name: Compress
+      if: runner.os == 'Windows'
+      shell: pwsh
+      run: |
+        cd build && Compress-Archive * ../Celeste64-Fuji-${{ matrix.rid }}.zip
+
+    - name: Compress
       if: runner.os != 'Windows'
+      run: |
+        cd build && zip -r ../Celeste64-Fuji-${{ matrix.rid }}.zip *
+
+    - name: Upload artifact to release
+      uses: ncipollo/release-action@v1
+      with:
+        artifacts: Celeste64-Fuji-${{ matrix.rid }}.zip
+        commit: "${{ github.sha }}"
+        tag: ${{ needs.release_info.outputs.version }}
+        allowUpdates: true
+        omitNameDuringUpdate: true
+        omitBodyDuringUpdate: true
+
+  publish:
+    name: Publish release
+    needs: [release_info, build]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Publish Github Release
+        uses: ncipollo/release-action@v1
+        with:
+          tag: ${{ needs.release_info.outputs.version }}
+          commit: "${{ github.sha }}"
+          allowUpdates: true
+          omitBodyDuringUpdate: true
+          omitNameDuringUpdate: true
+          draft: false
```

Comment:
I think we'll want to release it as a draft so we can review it and test it and make finishing touches to the release in github before actually publishing it. So we might not need this publishing step?

Reply:
Ah ok - makes sense!